compiler/luci/pass/src/ForwardTransposeOpPass.cpp

```diff
@@ -243,84 +240,28 @@ class EBOWithConstPattern final : public CircleNodeMutableVisitor<bool>
       return true;
     }
 
-// TODO Remove unused code
-#if 0
-    if (auto x = dynamic_cast<luci::CircleConst *>(node->x()))
-    {
-      if (auto y = dynamic_cast<luci::CircleTranspose *>(node->y()))
-      {
-        RETURN_FALSE_UNLESS(node->rank() == y->rank());
-        RETURN_FALSE_UNLESS(check_rank_four(x));
-        RETURN_FALSE_UNLESS(check_perm(y));
-
-        auto new_const = gen_new_const(y, x);
-        assert(new_const); // FIX_ME_UNLESS
-
-        auto new_transpose = create_cloned_transpose(y);
-        assert(new_transpose); // FIX_ME_UNLESS
-
-        // Reconnect network
-        node->x(new_const);
-        node->y(y->a());
-        loco::replace(node).with(new_transpose);
-        new_transpose->a(node);
-
-        // Do shape inference for this node again.
-        node->shape_status(luci::ShapeStatus::UNDEFINED);
-
-        return true;
-      }
-    }
-
-    if (auto y = dynamic_cast<luci::CircleConst *>(node->y()))
-    {
-      if (auto x = dynamic_cast<luci::CircleTranspose *>(node->x()))
-      {
-        RETURN_FALSE_UNLESS(node->rank() == x->rank());
-        RETURN_FALSE_UNLESS(check_rank_four(y));
-        RETURN_FALSE_UNLESS(check_perm(x));
-
-        auto new_const = gen_new_const(x, y);
-        assert(new_const); // FIX_ME_UNLESS
-
-        auto new_transpose = create_cloned_transpose(x);
-        assert(new_transpose); // FIX_ME_UNLESS
-
-        // Reconnect network
-        node->y(new_const);
-        node->x(x->a());
-        loco::replace(node).with(new_transpose);
-        new_transpose->a(node);
-
-        // Do shape inference for this node again.
-        node->shape_status(luci::ShapeStatus::UNDEFINED);
-
-        return true;
-      }
-    }
-#endif
-
     return false;
   }
 
 public:
   // Default
   bool visit(luci::CircleNode *) { return false; }
 
-  bool visit(luci::CircleAdd *node) { return has_pattern(node); }
+  bool visit(luci::CircleAdd *node) { return has_commutative_xy(node); }
 
-  bool visit(luci::CircleMul *node) { return has_pattern(node); }
+  bool visit(luci::CircleMul *node) { return has_commutative_xy(node); }
 
 private:
   // Return a new const node after Tranpose Op is forwarded
-  // Return nullptr if unsupported cases
+  // Never return nullptr
   CircleConst *gen_new_const(CircleTranspose *t, CircleConst *c)
```

Comment:
I made this function never return nullptr. Why? This function creates a new node, so all necessary checks have to be done before this function is invoked.